src/Database/Query.php

```diff
@@ -369,6 +372,58 @@ public function traverseParts(callable $visitor, array $parts)
         return $this;
     }
 
+    /**
+     * Creates a common table expression (CTE).
+     *
+     * @param string $name The CTE name.
+     * @param \Cake\Database\ExpressionInterface|\Closure|string $query The CTE query definition.
+     * @return \Cake\Database\Expression\CommonTableExpression
+     */
+    public function commonTableExpression(string $name, $query): CommonTableExpression
```

Comment:
I'm thinking of the complex SQL statements that would use this. I think `cte()` might be much cleaner even if it's less verbose.

Reply:
That's two votes for CTE then...

Reply:
I don't mind the verbosity as it is not much interface. Only if the interface is huge long names become an issue imho as they clump up app source everywhere.

Reply:
done